### ISSUE TITLE
[number field] min/max attributes

### ISSUE THREAD

**Issue by cgatian** [mui/base-ui#4226]
# Get help

## Ask a question

I expected the min and max props on a `NumberField` to set the `input`'s `min` and `max` attributes for screen readers. A suppose this is due to `type="text"` being used for the input.

I suppose the best way to provide accessiblity would be to provide a aria-description for screen readers? Any insights would help. Thank you!

**Comment by cgatian** [comment]
Im starting to think the role of the input should be a `spinbutton` which would support `aria-valuemin` ` aria-valuemax`. 
The current role is `textbox` with `aria-roledescription="Number field"`. Which probably doesnt translate for i18n. Curious what the maintainers think.

**Comment by cgatian** [comment]
OK our team went down a rabbit hole on this one. The Base UI implementation looks similar to [React Aria's implementation](https://react-aria.adobe.com/NumberField#value-scale)

Which they blogged about here:
https://react-aria.adobe.com/blog/how-we-internationalized-our-numberfield

And React Aria's implementation was based off this blog post by uk.gov
https://technology.blog.gov.uk/2020/02/24/why-the-gov-uk-design-system-team-changed-the-input-type-for-numbers/

Which provides really good insights into their a11y testing. Still digging to see if they have any guidance on min/max.

I will point out to maintainers that the `aria-roledescription` should potentially be called out that it should be translated for i18n support. React Aria also makes the `aria-roledescription` null for iOS devices:

https://github.com/adobe/react-spectrum/blob/d974b93dee0ba4dc7906095097348220ef778e99/packages/%40react-aria/numberfield/src/useNumberField.ts#L272
